File: common/src/main/java/com/yu/lib/common/ui/adapter/multi/BaseItemBinder.java
package com.yu.lib.common.ui.adapter.multi;

import android.content.Context;
import android.view.View;

public abstract class BaseItemBinder<T> {
    private String mViewTypeStr;
    int viewType;

    public String getViewTypeStr() {
        return mViewTypeStr;
    }

    void setViewTypeStr(String viewTypeStr) {
        mViewTypeStr = viewTypeStr;
    }

    public abstract int getLayoutRes();

    public View getView(Context context, int viewType) {
        return null;
    }

    public void onCreateView(MultiRecyclerAdapter.BaseViewHolder holder) {

    }

    public abstract void initBindData(MultiRecyclerAdapter.BaseViewHolder holder, int position, T t);

    public void bindData(MultiRecyclerAdapter.BaseViewHolder holder, int position, T t) {
    }

    public void onViewRecycled(MultiRecyclerAdapter.BaseViewHolder holder) {
    }

    public void onViewAttachedToWindow(MultiRecyclerAdapter.BaseViewHolder holder) {
    }

    public void onViewDetachedFromWindow(MultiRecyclerAdapter.BaseViewHolder holder) {
    }
}

File: common/src/main/java/com/yu/lib/common/ui/adapter/multi/BaseItemBinder.kt
package com.yu.lib.common.ui.adapter.multi

import android.content.Context
import android.view.View

abstract class BaseItemBinder<T> {
    private var mViewTypeStr: String? = null
    var viewType = 0

    open fun getViewTypeStr(): String? {
        return mViewTypeStr
    }

    open fun setViewTypeStr(viewTypeStr: String?) {
        mViewTypeStr = viewTypeStr
    }

    abstract fun getLayoutRes(): Int

    open fun getView(context: Context?, viewType: Int): View? {
        return null
    }

    abstract fun onCreateViewHolder(holder: MultiRecyclerAdapter<T>.BaseViewHolder)
    abstract fun onBindViewHolder(holder: MultiRecyclerAdapter<T>.BaseViewHolder, position: Int, data: T)

    open fun onViewRecycled(holder: MultiRecyclerAdapter<T>.BaseViewHolder) {}

    open fun onViewAttachedToWindow(holder: MultiRecyclerAdapter<T>.BaseViewHolder) {}

    open fun onViewDetachedFromWindow(holder: MultiRecyclerAdapter<T>.BaseViewHolder) {}
}